stdlib/src/collections/list.mojo

```diff
@@ -826,7 +929,9 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
         )
         return (self[].data + idx_as_int)[]
 
-    fn count[T: ComparableCollectionElement](self: List[T], value: T) -> Int:
+    fn count[
+        U: ComparableCollectionElement
+    ](self: Reference[List[U, Self.small_buffer_size], _, _], value: U) -> Int:
```

Comment:
I had to use references in those situations otherwise the compiler wasn't happy at all with the extra parameter. The behavior didn't change. Same for other methods like `__str__`

Reply:
Can you file compiler bugs please?  Your PRs are definitely hitting a few and I want to make sure they're all captured!  Perhaps a comment in the code too referencing such bugs?

Reply:
I don't mind filing those bugs but can it be after the PR is merged? Making a minimal reproducible example can sometimes take a few minutes but it can also take a lot of time. I can make another PRs to add the TODOs later if that's okay with you.

Reply:
Of course, totally fine.